libtextsecure/message_receiver.js

```diff
@@ -730,9 +730,18 @@ MessageReceiver.prototype.extend({
       case textsecure.protobuf.Envelope.Type.PREKEY_BUNDLE:
         window.log.info('prekey message from', this.getEnvelopeId(envelope));
         promise = captureActiveSession(sessionCipher)
-          .then(() =>
-            this.decryptPreKeyWhisperMessage(ciphertext, sessionCipher, address)
-          )
+          .then(async () => {
+            const buffer = dcodeIO.ByteBuffer.wrap(ciphertext);
+            await window.libloki.storage.verifyFriendRequestAcceptPreKey(
```

Comment:
Down on line 843
```
if (error && error.message === 'Unknown identity key') {
  // create an error that the UI will pick up and ask the
  // user if they want to re-negotiate
  const buffer = dcodeIO.ByteBuffer.wrap(ciphertext);
  errorToThrow = new textsecure.IncomingIdentityKeyError(
    address.toString(),
    buffer.toArrayBuffer(),
    error.identityKey
  );
} else {
  // re-throw
  throw error;
}
const ev = new Event('error');
ev.error = errorToThrow;
ev.proto = envelope;
ev.confirm = this.removeFromCache.bind(this, envelope);
```
I think this means that we won't be removing these messages from the cache unless I'm misunderstanding?
I.e. if you get one of these errors and then restart the app it will log the error again (probably 3 times total at which point it will be removed from the cache automatically)

Reply:
Good point, removing from cache.